tulip/transys/transys.py

```diff
@@ -786,3 +789,165 @@ def __init__(self):
             'type?label': '',
             'separator': r'\\n'}
         self.dot_node_shape = {'normal': 'rectangle'}
+
+
+def simu_abstract(ts,simu_type):
+    """Create a bi/dual-simulation abstraction for a Finite Transition System.
+    
+    @param ts: input finite transition system, the one you want to get 
+                    its bi/dual-simulation abstraction.
+    @type ts: L{FTS}
+    @param simu_type: string 'bi'/'dual', flag used to switch b.w. 
+                      bisimulation algorithm and dual-simulation algorithm.
+    @return: the bi/dual simulation, and the corresponding partition.
+    @rtype: L{FTS}, C{dict}
+    
+    References
+    ==========
+    1. Wagenmaker, A. J.; Ozay, N.
+    A Bisimulation-like Algorithm for Abstracting Control Systems. 
+    54th Annual Allerton Conference on CCC 2016
+    """
+    def pre(graph,list_n):
+        """Find union of predecessors of nodes in a graph or FTS.
+        
+        @param graph: a graph structure corresponding to a FTS.
+        @type graph: C{networkx.MultiDiGraph}
+        @param list_n: list of nodes whose pre's need to be returned
+        @type list_n: list of nodes in C{graph}
+        @return: set of predecessors of C{list_n}
+        @rtype: C{set}
+        """
+        # The simplest pre is implemented. Modify this fun to ContPre if
+        #  necessary.
+        pre_set = set()
+        for n in list_n:
+            pre_set = pre_set.union(graph.predecessors(n))
+        return pre_set
+    def output_FTS(ts,Part):
```

Comment:
Similarly to `pre`, the function `output_FTS` can be moved outside of `simu_abstract`, and renamed to `_output_fts`, in order to hide it and following PEP 8 ([snake_case](https://en.wikipedia.org/wiki/Snake_case)).